app/Services/AdventuresLeagueAdapter.php

```diff
@@ -153,12 +127,43 @@ private function getEntries($data, $characterId)
                     'description' => (array_key_exists(6, $data[$i])) ? $data[$i][6] : "",
                     'author_id' => Auth::id(),
                     'tier' => 0,
-                    'counted' => 0,
                 ];
-                $item = new Item($itemData);
-                $item->save();
+                Item::create($itemData);
             }
         }
         return $entries;
     }
+
+    private function getEntryData($characterId, $data, $type)
+    {
+        $entryData = [
+            'user_id' => Auth::id(),
+            'adventure_id' => null,
+            'campaign_id' => null,
+            'character_id' => $characterId,
+            'event_id' => null,
+            'dungeon_master_id' => null,
+            'levels' => 0,
+            ];
+
+        if (is_null($data)) {
+            $defaultEntryData = [
+                'date_played' => now(),
+                'gp' => 0,
+                'downtime' => 0,
+                'type' => Entry::TYPE_GAME,
+            ];
+            return array_merge($entryData, $defaultEntryData);
+        } else {
+            $populatedEntryData = [
+                'dungeon_master' => (array_key_exists(12, $data)) ? (float)$data[12] : "",
+                'location' => (array_key_exists(11, $data)) ? (float)$data[11] : "",
```

Comment:
wait are these supposed to be cast to a float? dungeon master and location are supposed to be a strings describing either item

Reply:
Is dungeon master the ID of the dungeon master or a string to refer to that person (maybe not their actual name)? I also remember the D&D MTL talking about possibly not having accounts for someone acting as the DM.

Reply:
> Is dungeon master the ID of the dungeon master or a string to refer to that person

In this case its going to be their name as a string

> I also remember the D&D MTL talking about possibly not having accounts for someone acting as the DM.

Right, that's why in the database we have 2 fields, if the user is not present in our system then they can enter the string representation of the DMs name

Reply:
not sure why these are being cast to float here, as you said they should be strings

Reply:
fixed.